services/rollout-service/pkg/versions/versions.go

```diff
@@ -218,21 +234,53 @@ func (v *versionClient) ConsumeEvents(ctx context.Context, processor VersionEven
 							continue
 						}
 
-						processor.ProcessKuberpultEvent(ctx, KuberpultEvent{
-							Application:      app.Name,
-							Environment:      env.Name,
-							EnvironmentGroup: envGroup.EnvironmentGroupName,
-							Team:             tm,
-							IsProduction:     env.Priority == api.Priority_PROD,
-							Version: &VersionInfo{
-								Version:        app.Version,
-								SourceCommitId: sc,
-								DeployedAt:     dt,
-							},
-						})
+						if v.manageArgoAppsEnabled && len(v.manageArgoAppsFilter) > 0 && strings.Contains(v.manageArgoAppsFilter, app.Name) {
+							manifestPath := path.Join("environments", env.Name, "applications", app.Name, "manifests", "manifests.yaml")
+							l.Info(manifestPath)
+
+							var annotations map[string]string
+							var labels map[string]string
+
+							for k, v := range env.Config.Argocd.ApplicationAnnotations {
+								annotations[k] = v
+							}
+							annotations["com.freiheit.kuberpult/team"] = tm
+							annotations["com.freiheit.kuberpult/application"] = app.Name
+							annotations["com.freiheit.kuberpult/environment"] = env.Name
+							annotations["com.freiheit.kuberpult/self-managed"] = "true"
+							annotations["com.freiheit.kuberpult/self-managed-filter"] = v.manageArgoAppsFilter
+							// This annotation is so that argoCd does not invalidate *everything* in the whole repo when receiving a git webhook.
+							// It has to start with a "/" to be absolute to the git repo.
+							// See https://argo-cd.readthedocs.io/en/stable/operator-manual/high_availability/#webhook-and-manifest-paths-annotation
+							annotations["argocd.argoproj.io/manifest-generate-paths"] = "/" + manifestPath
+							labels["com.freiheit.kuberpult/team"] = tm
+
+							deployApp := argo.CreateDeployApplication(overview, app, env, annotations, labels, manifestPath)
+
+							v.ArgoProcessor.Push(overview, deployApp)
```

Comment:
You only need to push the overview here. Also you are pushing for every application which is definitely too much.